scripts/create_repository.py

```diff
@@ -0,0 +1,198 @@
+import time
+import hashlib
+import json
+import ast
+import copy
+import glob
+import os
+
+root_scala_versions = ["2.11.12", "2.12.19", "2.13.14", "3.1.3", "3.2.2", "3.3.3", "3.4.3", "3.5.0"]
+scala_test_version = "3.2.9"
+scala_fmt_version = "3.0.0"
+
+@dataclass
+class MavenCoordinates:
+  group: str
+  artifact: str
+  version: str
+  coordinate: str
+
+@dataclass
+class ResolvedArtifact:
+  coordinates: MavenCoordinates
+  checksum: str
+  direct_dependencies: List[MavenCoordinates]
+
+def select_root_artifacts(scala_version) -> List[str]:
+  scala_major = ".".join(scala_version.split(".")[:2])
+  scala_test_major = "3" if scala_major >= "3.0" else scala_major
+  scala_fmt_major = "2.13" if scala_major >= "3.0" else scala_major
+  kind_projector_version = "0.13.2" if scala_major < "2.13" else "0.13.3"
+
+  common_root_artifacts = [
+    f"org.scalatest:scalatest_{scala_test_major}:{scala_test_version}",
+    f"org.scalameta:scalafmt-core_{scala_fmt_major}:{"2.7.5" if scala_major == "2.11" else scala_fmt_version}"
+  ]
+
+  scala_artifacts = [
+    f'org.scala-lang:scala3-library_3:{scala_version}',
+    f'org.scala-lang:scala3-compiler_3:{scala_version}',
+    f'org.scala-lang:scala3-interfaces:{scala_version}',
+    f'org.scala-lang:tasty-core_3:{scala_version}'
+  ] if scala_major[0] == "3" else [
+        f'org.scala-lang:scala-library:{scala_version}',
+        f'org.scala-lang:scala-compiler:{scala_version}',
+        f'org.scala-lang:scala-reflect:{scala_version}',
+        f'org.scalameta:semanticdb-scalac_{scala_version}:4.9.9',
+        f'org.typelevel:kind-projector_{scala_version}:{kind_projector_version}'
+  ]
+
+  return common_root_artifacts + scala_artifacts
+
+def get_maven_coordinates(artifact) -> MavenCoordinates:
+  splitted = artifact.split(':')
+  version = splitted[2] if splitted[2][0].isnumeric() else splitted[3]
+  return MavenCoordinates(splitted[0], splitted[1], version, artifact)
+
+def get_mavens_coordinates_from_json(artifacts) -> List[MavenCoordinates]:
+  return list(map(lambda artifact: get_maven_coordinates(artifact), artifacts))
+
+def get_artifact_checksum(artifact) -> str:
+    output = subprocess.run(f'cs fetch {artifact}', capture_output=True, text=True, shell=True).stdout.splitlines()
+    possible_url = [o for o in output if "https" in o][0]
+    possible_url = possible_url[possible_url.find("https"):].replace('https/', 'https://')
+    try:
+        with urllib.request.urlopen(possible_url) as value:
+            body = value.read()
+            return hashlib.sha256(body).hexdigest()
+    except urllib.error.HTTPError as e:
+      print(f'RESOURCES NOT FOUND: {possible_url}')
+
+def get_json_dependencies(artifact) -> List[MavenCoordinates]:
+  with open('out.json') as file:
+    data = json.load(file)
+    return get_mavens_coordinates_from_json(dependency["directDependencies"]) if any((dependency := d)["coord"] == artifact for d in data["dependencies"]) else []
+
+def get_label(coordinate) -> str:
```

Comment:
There are 5 patterns of labels, which are related to the structure of certain group and artifact due to custom naming.